indico/modules/rb_new/client/js/modules/admin/EditableList.jsx

```diff
@@ -0,0 +1,173 @@
+                <Modal.Header>
+                    {title}
+                </Modal.Header>
+                <Modal.Content>
+                    <FinalForm onSubmit={this.handleSubmit}
+                               initialValues={initialValues}
+                               initialValuesEqual={_.isEqual}
+                               subscription={{submitting: true, hasValidationErrors: true, pristine: true}}>
+                        {fprops => (
+                            <Form onSubmit={fprops.handleSubmit}>
+                                {renderForm(fprops)}
+                                <Form.Button type="submit"
+                                             disabled={(
+                                                 fprops.hasValidationErrors ||
+                                                 fprops.pristine ||
+                                                 fprops.submitting
+                                             )}
+                                             loading={fprops.submitting}
+                                             primary
+                                             content={Translate.string('Add')} />
+                            </Form>
+                        )}
+                    </FinalForm>
+                </Modal.Content>
+            </Modal>
+        );
+    }
+}
+
+/**
+ * A list component that supports adding, removing, and inline editing.
+ */
+export default class EditableList extends React.PureComponent {
+    static propTypes = {
+        title: PropTypes.string.isRequired,
+        addModalTitle: PropTypes.string.isRequired,
+        isFetching: PropTypes.bool.isRequired,
+        items: PropTypes.array.isRequired,
+        initialAddValues: PropTypes.object,
+        initialEditValues: PropTypes.func,
+        renderItem: PropTypes.func.isRequired,
+        renderAddForm: PropTypes.func.isRequired,
+        renderEditForm: PropTypes.func.isRequired,
+        renderDeleteMessage: PropTypes.func.isRequired,
+        onDelete: PropTypes.func.isRequired,
+        onUpdate: PropTypes.func.isRequired,
+        onCreate: PropTypes.func.isRequired,
+    };
+
+    static defaultProps = {
+        initialAddValues: {},
+        initialEditValues: undefined,
+    };
+
+    state = {
+        adding: false,
+    };
+
+    handleAddClick = () => {
+        this.setState({adding: true});
+    };
+
+    handleAddClose = () => {
+        this.setState({adding: false});
+    };
+
+    handleCreate = async (data) => {
+        const {onCreate} = this.props;
+        const rv = await onCreate(data);
+        if (!rv.error) {
+            this.setState({adding: false});
+        }
+        return rv;
+    };
+
+    render() {
+        const {
+            title, addModalTitle, renderAddForm, renderEditForm, renderDeleteMessage, renderItem, initialEditValues,
+            isFetching, items,
+            initialAddValues, onDelete, onUpdate,
```

Comment:
Can't it appear on the previous line with `isFetching, items`?

Reply:
Initially I had only the actions on that line, but yes, can be merged now